File: back_end/src/main/java/kr/co/has/hygiene/back_end/domain/RecordingPaper.java
package kr.co.has.hygiene.back_end.domain;


import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Entity
public class RecordingPaper {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    Long id;
    String observeDepartment;//부서
    String observeOccupation;//직종
    String observeName; //성명
    String location; //관찰장소
    String action; //행위
    String subAction; //세부행위
    Boolean glove; //장갑
    LocalDateTime actionStartTime; //시작시간
//    LocalDateTime actionEndTime;
    Integer actionDuration; //수행시간
    String actionType; //수행여부 - 손소독,손세척,미수행.
    Boolean passFail; //수행적합성
    String observer; //관찰자

}

File: front_end/src/has/components/Types/RecordingPaper.ts
import dayjs, {Dayjs} from "dayjs";

export enum E_chronometer{
    Ready,
    Run,
    Stop
}
export class RecordingPaper {
    id?: string;
    observeDepartment?: string;
    observeOccupation?: string;
    observeName?: string;
    location?: string;
    action?: string;
    subAction?: string;
    glove?: boolean = false;
    actionStartTime?: Dayjs|string;
    // actionEndTime?: Dayjs|string;
    actionDuration :number = 0;
    actionType?: string;
    passFail?: boolean = false;
    observer?: string;

    key?: string;
    chronometer_state?: E_chronometer = E_chronometer.Ready;

    public _현재가능액션():string{
        if(this.chronometer_state == E_chronometer.Ready){
            return "시작";

        }

        if(this.chronometer_state == E_chronometer.Run){
            return "중지";
        }
        return "완료";
    }
    public obtain수행시간():number{
        if(this.chronometer_state == E_chronometer.Ready)
            return 0;

        if(this.chronometer_state == E_chronometer.Run) {
            this.actionDuration = Math.trunc(dayjs().diff(this.actionStartTime) / 1000);
            return this.actionDuration;
        }

        // E_chronometer.Stop
        return this.actionDuration;
    }
    public nextStateClickChronometer(){
        if(this.chronometer_state == E_chronometer.Ready){
            this.actionStartTime = dayjs();
            this.chronometer_state = E_chronometer.Run;
            return;
        }

        if(this.chronometer_state == E_chronometer.Run){
            this.actionDuration = Math.trunc(dayjs().diff(this.actionStartTime) / 1000);
            this.chronometer_state = E_chronometer.Stop;
            return;
        }

        return;
    }
}
